library/aes.c

```diff
@@ -662,6 +670,8 @@ int mbedtls_aes_setkey_dec( mbedtls_aes_context *ctx, const unsigned char *key,
     uint32_t *RK;
     uint32_t *SK;
 
+    MBEDTLS_AES_VALIDATE_RET( ctx != NULL && key != NULL );
```

Comment:
I would suggest splitting this condition into two separate checks. Otherwise, it is a bit annoying to having to work out which branch of the && failed